// Parser Interface v4.0

syntax = "proto3";

message ATRResponse {
  optional bytes raw_atr = 1;
  optional string chip = 2;
  optional string standard = 3;
  optional string file_structure = 4;
  optional string software_issuer = 5;
  optional string software_version = 6;
  optional string lower_serial = 7;
}

message Command {
  optional uint32 pcap_index = 1;
  optional uint32 cla = 2;
  optional uint32 ins = 3;
  optional uint32 p1 = 4;
  optional uint32 p2 = 5;
  optional uint32 lc = 6;
  optional bytes data = 7;
  optional uint32 le = 8;
  optional bool is_known_command = 9;
  optional string description = 10;
}

message Response {
  optional uint32 pcap_index = 1;
  optional bytes data = 2;
  optional uint32 sw1 = 3;
  optional uint32 sw2 = 4;
  optional string status = 5;
  optional string description = 6;
}

message PcapParseResponse {
  optional bool status = 1;
  optional ATRResponse atr_response = 2;
  repeated Command commands = 3;
  repeated Response responses = 4;
}
